snuba/query/expressions.py

```diff
@@ -85,6 +86,32 @@ def functional_eq(self, other: Expression) -> bool:
         """
         raise NotImplementedError
 
+    # Sort by Class first and then alphabetically
+    # Class order: Literals < Columns < FunctionCalls (nested)
+    def compare_expressions(
+        self, other: Expression, op: Callable[[Any, Any], Any]
+    ) -> bool:
+        relate = {operator.lt: True, operator.gt: False}
+
```

Comment:
To your points:

1. When only one side is a Column, this would imply that the other side is a literal (at least in the test cases). The recursive `.compare_expressions()` is called and orders them accordingly. However, this implication might not always be true outside of test cases and if we decide to support more types in the future. I will add that in.

2. When comparing two FunctionCalls, we're actually comparing the first two objects in the first index of the tuple. For example:
* A = FunctionCall([Column('b') = 1, Column('d') = 1])
* B = FunctionCall([Column('a') = 1, Column('c') = 1])

In this case, B < A because Column('a') is < Column('b').
My initial thought was to keep this logic outside of FunctionCall and put it in the parent class for isolation. I agree that it makes more sense for the parent function to solely be used to compare two different types. However, this requires `FunctionCall` to know about its sibling classes `Column` and `Literal`. I've made this change to reflect this.